### PR TITLE
[sail] Update to 0.9.0-pre16

### PR DESCRIPTION
**Pull request by HappySeaFox**
**Describe the pull request**

Updates in the upstream, including a couple of memory leaks.

- #### What does your PR fix?  
  \-

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  Same as before

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  `Yes`

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/sail/portfile.cmake [hunk 1]
```diff
@@ -3,8 +3,8 @@ vcpkg_fail_port_install(ON_TARGET "UWP")
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO smoked-herring/sail
```
Comment (PhoebeHui) [inline ports/sail/portfile.cmake:5]:
```suggestion
    REPO HappySeaFox/sail
```

**Comment by PhoebeHui** [comment]
@HappySeaFox, thanks for the PR!

**Review by PhoebeHui — APPROVED** [review]
LGTM, thanks for your updates!

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**